Admin Console uses a wrong URL type for auth server

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

The Admin Console seems to be using the admin URL type instead of frontend URL type when configuring auth server URL in the JS adapter.

### Version

main

### Regression

- [X] The issue is a regression

### Expected behavior

When not logged in, the Admin Console should redirect the user to the login page using frontend URL.

### Actual behavior

When not logged in, the Admin Console redirects the user to the login page using admin URL.

### How to Reproduce?

1. Start the server as:
```
kc.sh start-dev --hostname=http://127.0.0.1.nip.io:8080 --hostname-admin=http://admin.127.0.0.1.nip.io:8080
```
2. Access: http://127.0.0.1.nip.io:8080/admin
3. Browser is redirected to login page at `http://admin.127.0.0.1.nip.io:8080/realms/master/protocol/openid-connect/auth?...`

This behaviour is incorrect. Login flow needs to be always executed at frontend URL, in this case `http://127.0.0.1.nip.io:8080`.

### Anything else?

This caused by https://github.com/keycloak/keycloak/commit/e28aa90fcb6e2603ec1222ca01da18dcde7ede2d.

it doesn't seem to start with this command:
```
2024-05-17 14:43:37,226 INFO  [org.infinispan.CONTAINER] (ForkJoinPool.commonPool-worker-1) ISPN000556: Starting user marshaller 'org.infinispan.jboss.marshalling.core.JBossUserMarshaller'
2024-05-17 14:43:37,575 INFO  [org.keycloak.broker.provider.AbstractIdentityProviderMapper] (main) Registering class org.keycloak.broker.provider.mappersync.ConfigSyncEventListener
2024-05-17 14:43:37,749 INFO  [org.keycloak.connections.infinispan.DefaultInfinispanConnectionProviderFactory] (main) Node name: node_339315, Site name: null
2024-05-17 14:43:38,117 INFO  [io.quarkus] (main) Keycloak 999.0.0-SNAPSHOT on JVM (powered by Quarkus 3.8.4) started in 2.643s. Listening on: http://0.0.0.0:8080. Management interface listening on http://0.0.0.0:9000.
2024-05-17 14:43:38,117 INFO  [io.quarkus] (main) Profile dev activated.
2024-05-17 14:43:38,117 INFO  [io.quarkus] (main) Installed features: [agroal, cdi, hibernate-orm, jdbc-h2, keycloak, logging-gelf, narayana-jta, reactive-routes, resteasy-reactive, resteasy-reactive-jackson, smallrye-context-propagation, vertx]
2024-05-17 14:43:38,121 WARN  [org.keycloak.quarkus.runtime.KeycloakMain] (main) Running the server in development mode. DO NOT use this configuration in production.
2024-05-17 14:43:55,888 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-1) Uncaught server error: jakarta.ws.rs.WebApplicationException: HTTP 500 Internal Server Error
	at org.keycloak.services.resources.WelcomeResource.createWelcomePage(WelcomeResource.java:235)
	at org.keycloak.services.resources.WelcomeResource.getWelcomePage(WelcomeResource.java:95)
	at org.keycloak.services.resources.WelcomeResource$quarkusrestinvoker$getWelcomePage_3d46f4348f9016709ff9c5ca428ac29847ab83d6.invoke(Unknown Source)
	at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:582)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1589)
Caused by: jakarta.ws.rs.core.UriBuilderException: failed to create URI
	at org.jboss.resteasy.reactive.common.jaxrs.UriBuilderImpl.buildFromValues(UriBuilderImpl.java:753)
	at org.jboss.resteasy.reactive.common.jaxrs.UriBuilderImpl.build(UriBuilderImpl.java:741)
	at org.keycloak.models.KeycloakUriInfo.getBaseUri(KeycloakUriInfo.java:98)
	at org.keycloak.models.KeycloakUriInfo.getBaseUriBuilder(KeycloakUriInfo.java:105)
	at org.keycloak.services.resources.WelcomeResource.createWelcomePage(WelcomeResource.java:188)
	... 12 more
Caused by: java.net.URISyntaxException: Expected scheme-specific part at index 5: http:
	at java.base/java.net.URI$Parser.fail(URI.java:2970)
	at java.base/java.net.URI$Parser.failExpecting(URI.java:2976)
	at java.base/java.net.URI$Parser.parse(URI.java:3176)
	at java.base/java.net.URI.<init>(URI.java:620)
	at org.jboss.resteasy.reactive.common.jaxrs.UriBuilderImpl.buildFromValues(UriBuilderImpl.java:748)
	... 16 more
```

--- a/js/libs/ui-shared/src/context/KeycloakContext.tsx
+++ b/js/libs/ui-shared/src/context/KeycloakContext.tsx
@@ -49,7 +49,10 @@ export const KeycloakProvider = <T extends BaseEnvironment>({
   const [error, setError] = useState<unknown>();
   const keycloak = useMemo(() => {
     const keycloak = new Keycloak({
-      url: environment.authUrl,
+      url:
+        "authServerUrl" in environment
+          ? (environment.authServerUrl as string)
+          : environment.authUrl,
       realm: environment.realm,
       clientId: environment.clientId,
     });

